cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb

```diff
@@ -16,26 +16,64 @@ module Algorithms
   HOUR_IN_SECONDS = 3600
 
   def init_jwks_file(file_name)
-    jwks = { keys: [jwk.export] }
+    add_new_file(file_name)
+    jwks = { keys: [jwk_set[file_name].export] }
     File.write(
       "#{JWKS_ROOT_PATH}/#{file_name}",
       JSON.pretty_generate(jwks)
     )
   end
 
+  def init_second_jwks_file_with_same_kid(first_file_name, second_file_name)
+    add_new_file(second_file_name)
+    jwk = jwk_set[second_file_name].export
+    jwk[:kid] = jwk_set[first_file_name].export[:kid]
+    jwks = { keys: [jwk] }
+
+    File.write(
+      "#{JWKS_ROOT_PATH}/#{second_file_name}",
+      JSON.pretty_generate(jwks)
+    )
+  end
+
   def issue_jwt_token(token_body, algorithm = Algorithms::RS256)
     @jwt_token = JWT.encode(
       token_body,
-      rsa_key,
+      rsa_keys.values[0],
+      algorithm,
+      { kid: jwk_set.values[0].kid }
+    )
+  end
+
+  def issue_jwt_token_with_jku(token_body, file_name, algorithm = Algorithms::RS256)
+    @jwt_token = JWT.encode(
+      token_body,
+      rsa_keys[file_name],
+      algorithm,
+      {
+        kid: jwk_set.values[0].kid,
+        jku: "#{JWKS_ROOT_PATH}/#{file_name}"
+      }
+    )
+  end
+
+  def issue_jwt_token_with_jwk(token_body, file_name, algorithm = Algorithms::RS256)
+    jwk = jwk_set[file_name].export
+    jwk[:kid] = jwk_set.values[0].kid
```

Comment:
I don't like `0` there. Can we fetch kid by name of the first file?

Reply:
It is possible but that means that we will need to pass the first filename to each of the functions which in my opinion is not so clean

Reply:
done